### PR TITLE
INT-5235 step: fetch-cloud-build-worker-pools

### PR DESCRIPTION
**Pull request by i5o**
Added support for Worker Pools, this introduces the need of specifying regions as the API requires it.
https://cloud.google.com/build/docs/api/reference/rest/v1/projects.locations.workerPools/list

### CODE REVIEW

File: src/steps/cloud-build/client.ts [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
 import { Client } from '../../google-cloud/client';
+import { CloudBuildLocations } from './constants';
 
 export class CloudBuildClient extends Client {
   private client = google.cloudbuild({ version: 'v1' });
 
+  private async iterateLocations(callback: (region: string) => Promise<void>) {
+    for (const region of CloudBuildLocations) {
+      await callback(region);
+    }
```
Comment (ndowmon) [inline src/steps/cloud-build/client.ts:11]: 🔥

File: src/steps/cloud-build/converters.ts [hunk 2]
```diff
@@ -56,11 +56,32 @@ export const createGoogleCloudBuildTriggerEntity = (
...
+      source: data,
+      assign: {
+        _class: CloudBuildEntitiesSpec.BUILD_WORKER_POOL._class,
+        _type: CloudBuildEntitiesSpec.BUILD_WORKER_POOL._type,
+        _key: data.name as string,
+        name: data.name,
+        displayName: data.displayName as string,
+        createdOn: parseTimePropertyValue(data.createTime),
+        updatedOn: parseTimePropertyValue(data.updateTime),
+        deletedOn: parseTimePropertyValue(data.deleteTime),
```
Comment (ndowmon) [inline src/steps/cloud-build/converters.ts:84]: Can you add `state` and `uid` as properties to the entity?

Also, can you please add the `annotations` as tags? TIA!

**Review by ndowmon — APPROVED** [review]
Approved with some comments about the properties to ingest. Thanks!

File: src/steps/cloud-build/converters.ts [hunk 3]
```diff
@@ -56,11 +56,32 @@ export const createGoogleCloudBuildTriggerEntity = (
...
+export const createGoogleCloudBuildWorkerPoolEntity = (
+  data: cloudbuild_v1.Schema$WorkerPool,
+) => {
+  return createGoogleCloudIntegrationEntity(data, {
+    entityData: {
+      source: data,
+      assign: {
+        _class: CloudBuildEntitiesSpec.BUILD_WORKER_POOL._class,
+        _type: CloudBuildEntitiesSpec.BUILD_WORKER_POOL._type,
+        _key: data.name as string,
```
Comment (ndowmon) [inline src/steps/cloud-build/converters.ts:79]: Actually, should we use `uid` here instead?

File: src/steps/cloud-build/converters.ts (hunk 2 above)
Comment (i5o) [reply src/steps/cloud-build/converters.ts:84]: sure thing

**Review by ndowmon — APPROVED** [review]

**Pull request merged**